image-loader/app/controllers/ImageLoaderController.scala

```diff
@@ -148,38 +155,58 @@ class ImageLoaderController(auth: Authentication,
       )
 
       logger.info("importImage request start")
+      val existingImageStatus = uploadStatusTable.getStatus(uri.split("/").last).map {
```

Comment:
Splitting the uri here is only going to work if the image ID is the last component which is not guaranteed. It would be much better to use the actual ID of the file which is available as the `digest` from the `digestedFile` in the for comprehension.